inc/namespace.php

```diff
@@ -196,3 +198,77 @@ function enqueue_scripts() {
 	// Print queued scripts.
 	print_head_scripts();
 }
+
+/**
+ * Schedule index maintenance daily.
+ */
+if( ! wp_next_scheduled( __NAMESPACE__ . '\\index_maintenance' ) ) {
+	wp_schedule_event( time(), 'daily',  __NAMESPACE__ . '\\index_maintenance' );
+}
+
+
+/**
+ * Delete old Analytics indexes in Elasticsearch.
+ */
+function delete_old_indexes() : ?array {
+	// Index age in days.
+	$duration = apply_filters( 'altis.analytics.index_age', 7 );
+
+	// If duration has been set out of the 60-day limit, default to 7 days and warn user.
+	if ( $duration < 1 || $duration > 60 ) {
+		$duration = 7;
+		trigger_error( 
+			'Analytics data retention period must be between 1 and 60 days, defaulting to 7.', 
```

Comment:
Rather than completely change the number of days we can move the retention period to the nearest value instead using `min()` and `max()`:
```suggestion
		$duration = max( 1, min( $duration, 60 ) );
		trigger_error( 
			sprintf( 'Analytics data retention period must be between 1 and 60 days. This has been set to %d.', $duration ), 
```